pkg/trace/api/listener_test.go

```diff
@@ -3,6 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
+// +build !race
```

Comment:
I believe that, instead of disabling the race detector, this disables the test when the race detector is active - not really what we want!

Reply:
>I believe that, instead of disabling the race detector, this disables the test when the race detector is active - not really what we want!
🤦 you're right, thats definitely not the way to go in how to fix this 

Thanks for spotting this. I'll close this PR, and discuss with @gbbr how to best fix this race condition. 

>most easily done with atomics, but sometimes a mutex/condvar are useful as well.
Yeah, I was thinking how to best approach it, - if we were to [synchronize the access to this variable](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/metrics/client.go#L29) there are some potential performance considerations. Not yet sure if they're acceptable, but probably an rw_mutex will not be half bad.

However there is also alternative to prevent this race condition from occurring in the testsuite - is to avoid `init()` starting [this metrics buffering ](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/metrics/timing/timing.go#L25) goroutine loop